### ISSUE TITLE
Blank Results but row count > 0

### ISSUE THREAD

**Issue by gstitan** [microsoft/azuredatastudio#25245]
Type: <b>Bug</b>

Run a simple Top n query: 
SELECT TOP 100 *
    , SysStartTime
FROM EncounterNote
ORDER BY id DESC

Status bar shows 100 rows returned but the Results are blank. 

If I click back and forth between Messages and Results, the Results will eventually show. 

Azure Data Studio version: azuredatastudio 1.47.1 (b6f7beb01f92adaa4b79b6b6f3ac704e95cafe6e, 2024-01-10T02:15:02.628Z)
OS version: Windows_NT x64 10.0.19044
Restricted Mode: No
Preview Features: Enabled
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD EPYC 7452 32-Core Processor                 (16 x 2346)|
|GPU Status|2d_canvas: unavailable_software<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: disabled_software<br>multiple_raster_threads: enabled_on<br>opengl: disabled_off<br>rasterization: disabled_software<br>raw_draw: disabled_off_ok<br>video_decode: disabled_software<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: unavailable_software<br>webgl2: unavailable_software<br>webgpu: unavailable_software|
|Load (avg)|undefined|
|Memory (System)|128.00GB (107.94GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (14)</summary>

Extension|Author (truncated)|Version
---|---|---
admin-pack|Mic|0.0.2
admin-tool-ext-win|Mic|0.1.3
agent|Mic|0.49.0
dacpac|Mic|1.14.0
import|Mic|1.6.0
profiler|Mic|0.14.0
query-history|Mic|0.5.3
schema-compare|Mic|1.21.0
sql-assessment|Mic|0.6.4
sql-migration|Mic|1.5.1
azdata-sanddance|msr|4.0.3
extra-sql-script-as|pac|0.5.0
sql-prompt|Red|0.2.14
poor-sql-formatter|WSR|0.1.0

</details>

**Comment by jlspake** [comment]
I am also having this issue. At first switching between Results and Messages would get the results to render, but after a few queries even that is no longer working for me. Only the header row is drawn; no values or even grid lines are drawn for the result rows.

**Comment by kisantia** [comment]
Did this only start happening with 1.47.1? Did 1.47 not have this issue?

**Comment by jlspake** [comment]
> Did this only start happening with 1.47.1? Did 1.47 not have this issue?

I think it existed before. I noticed that I had an update pending so I installed the update to see if that would fix the issue, but the issue remains. Unfortunately I did not check which version I was on before installing the update

**Comment by kisantia** [comment]
If it's a relatively recent problem that was introduced, the 1.47 added this change https://github.com/microsoft/azuredatastudio/pull/24719 that added an additional check for the formatting of the result cells, which could potentially have caused the slowdown. Can you try the [1.46 release](https://github.com/microsoft/azuredatastudio/releases/tag/1.46.1) and see if that's any better?

**Comment by jlspake** [comment]
Here's some additional info I just discovered while testing. 

In about half of cases I saw the results grid get drawn for a brief moment, and then it disappeared. I didn't see actual values drawn, but the grid lines flickered in and then vanished.

For me the problem seems to be query/data dependent - in the particular query I was running the results began showing as soon as I excluded one nvarchar(max) column from the data set. There were other nvarchar(max) columns included, but the others didn't cause a problem. Upon further investigation when I took a descending top 10 instead of ascending my query worked which seems to indicate there's something about the data itself that can't be rendered. My column is storing email message body data so sometimes plaintext and sometimes html

**Comment by khalabuza** [comment]
Very similar behavior for me. I am also storing email message strings (varchar (5000)) and if I exclude the column everything works fine. If included, random rows disappear, see below...

![image](https://github.com/microsoft/azuredatastudio/assets/93348869/ab521a05-e9d9-4eb6-a635-93d3bd33f0f3)

**Comment by sora-tech** [comment]
This was happening with 1.47.0 as I just upgraded to .1 to see if this was fixed.  I can reproduce with the following HTML every time:
```html
T<i style="">ex</i>t 3
```
Tabbing back and forth between results and messages works sometimes for now.

Other HTML seems ok, the following don't cause the issue:
```html
T<b>ext 2</b>
```
```html
<u style="">Text 1!</u>
```

Further investigation gives me this as always reproducible:
```SQL
SELECT 'a<i>b</i>c'
```
It doesn't matter what the element is, whenever there's text-element-text it breaks. Text-element or element-text are ok.
